File: 01-week-1--starter-algorithms/01-day-2--find-first-duplicate/javascript/find_first_duplicate.js
function findFirstDuplicate(arr) {
  // type your code here
  const unq = new Set()

  for (const value of arr){
    if (unq.has(value)){
      return value
    }
      unq.add(value)
  }
    return -1
}


if (require.main === module) {
  // add your own tests in here
  console.log("Expecting: 3");
  console.log("=>", findFirstDuplicate([2, 1, 3, 3, 2]));

  console.log("");

  console.log("Expecting: -1");
  console.log("=>", findFirstDuplicate([1, 2, 3, 4]));
}

module.exports = findFirstDuplicate;

// Please add your pseudocode to this file
// initalize an empty set called unq 

// iterate thru the input array 
//    if value is in unq, return value
//    else add value to the set
// return -1 if no duplicate found during iteration 

// And a written explanation of your solution

File: 01-week-1--starter-algorithms/01-day-2--find-first-duplicate/ruby/find_first_duplicate.rb
require 'set'

def find_first_duplicate(arr)
  uniques = Set.new 

  arr.each do |value| 
    return value if uniques.include?(value)
  end
    uniques.add(value)
  end
    -1 
end

if __FILE__ == $PROGRAM_NAME
  puts "Expecting: 3"
  puts "=>", find_first_duplicate([2, 1, 3, 3, 2])

  puts

  puts "Expecting: -1"
  puts "=>", find_first_duplicate([1, 2, 3, 4])

  # Don't forget to add your own!
end

# Please add your pseudocode to this file
# create empty set to hold values 
# iterate over array values
# if set includes value return value 
# if not, add it to the set 
# if no matches are found, return -1
# And a written explanation of your solution
